.github/workflows/CI.yml

```diff
@@ -5,26 +5,24 @@ on:
   push:
     branches: [ main, test_CI ]
 
+# TODO: create a game
 jobs:
   testbuild:
     runs-on: ubuntu-latest
     steps:
-    - name: "Checkout"
-      uses: actions/checkout@v2
+      - name: "Checkout"
+        uses: actions/checkout@v2
 
-    - name: "Setup Python"
-      uses: actions/setup-python@v2
-      with:
-        python-version: '3.7'
+      - name: "Setup Python"
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.7'
+      
+      - name: "Build Docker Container"
+        run: |
+          docker build --no-cache -t ci_image .
 
-    - name: "Build"
-      run: |
-        pip install -r requirements.txt
-        pip install -e .
-
-    - name: "Run tests"
-      run: |
-        pytest tests/utils_test.py
-        pytest tests/bot_tests/bots_test.py
-        # pytest tests/randomize_order_test.py
-        # pytest tests/bot_tests/soa_bot_functions_test.py
+      - name: "Run Tests"
+        run: |
+          docker run --rm ci_image /bin/bash -c "/model/src/model_server/baseline_bots/containers/allan_dip_bot/run_model_server.sh & \
+          pytest --ignore=tests/bot_tests/SOA_test.py"
```

Comment:
So we are running tests outside the container, correct?

Reply:
Sort of. Technically the tests run inside of the container. We are passing a container an instruction to run inside of itself.